crates/bevy_window/src/window.rs

```diff
@@ -103,35 +172,152 @@ pub enum CompositeAlphaMode {
+    /// Windows 11 is related to <https://github.com/rust-windowing/winit/issues/2082>
+    pub transparent: bool,
+    /// Should the window start focused?
+    pub focused: bool,
+    /// Should the window always be on top of other windows?
+    ///
+    /// ## Platform-specific
+    ///
+    /// - iOS / Android / Web / Wayland: Unsupported.
+    pub always_on_top: bool,
+    /// The "html canvas" element selector.
+    ///
+    /// If set, this selector will be used to find a matching html canvas element,
+    /// rather than creating a new one.   
+    /// Uses the [CSS selector format](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector).
+    ///
+    /// This value has no effect on non-web platforms.
+    pub canvas: Option<String>,
+    /// Whether or not to fit the canvas element's size to its parent element's size.
+    ///
+    /// **Warning**: this will not behave as expected for parents that set their size according to the size of their
+    /// children. This creates a "feedback loop" that will result in the canvas growing on each resize. When using this
+    /// feature, ensure the parent's size is not affected by its children.
+    ///
+    /// This value has no effect on non-web platforms.
+    pub fit_canvas_to_parent: bool,
+    /// Stores internal state that isn't directly accessible.
+    pub internal: InternalWindowState,
+}
+
+impl Default for Window {
+    fn default() -> Self {
+        Self {
+            title: "Bevy App".to_owned(),
+            cursor: Default::default(),
+            present_mode: Default::default(),
+            mode: Default::default(),
+            position: Default::default(),
+            resolution: Default::default(),
+            internal: Default::default(),
+            composite_alpha_mode: Default::default(),
+            resize_constraints: Default::default(),
+            resizable: true,
+            decorations: true,
+            transparent: false,
+            focused: true,
+            always_on_top: false,
+            fit_canvas_to_parent: false,
+            canvas: None,
+        }
     }
 }
 
-use crate::CursorIcon;
-use std::fmt;
+impl Window {
+    /// Setting this to true will attempt to maximize the window.
+    ///
+    /// Setting it to false will attempt to un-maximize the window.
+    pub fn set_maximized(&mut self, maximized: bool) {
+        self.internal.maximize_request = Some(maximized);
+    }
 
-use crate::raw_handle::RawHandleWrapper;
+    /// Setting this to true will attempt to maximize the window.
+    ///
+    /// Setting it to false will attempt to un-maximize the window.
+    pub fn set_minimized(&mut self, minimized: bool) {
+        self.internal.minimize_request = Some(minimized);
+    }
 
-impl fmt::Display for WindowId {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.0.as_simple().fmt(f)
+    /// The window's client area width in logical pixels.
+    #[inline]
+    pub fn width(&self) -> f32 {
```

Comment:
If we're going to abstract over width / height, we should probably abstract over the other resolution properties too

Reply:
Makes sense. We should definitely expose physical resolution properties too though.

Reply:
Ah true, overlooked that, I'll put those in